lib/fs.js

```diff
@@ -903,17 +903,27 @@ fs.mkdirSync = function(path, mode) {
                        modeNum(mode, 0o777));
 };
 
-fs.readdir = function(path, callback) {
+fs.readdir = function(path, options, callback) {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  if (typeof options !== 'object') {
```

Comment:
In that case this should be attached as an `else if` to the above, but also if options is truthy wouldn't that also do the wrong thing? Or, guess since "everything's an object" means as long as the `encoding` property's defined then we shouldn't care. okay. that solution is cool.